reference/extensions/deployers.rst

```diff
@@ -1,5 +1,65 @@
 .. _reference_extensions_deployers:
 
 Deployers
----------
+=========
 
+Deployers are a mechanism to facilitate copying files form one folder, usually the Conan cache, to user folders.
+While Conan provides two built-in ones (``full_deploy`` and ``direct_deploy``), users can easily manage their own
+with ``conan config install``.
+
+Deployers run before generators, and they can change the target folders.
+For example, if the ``--deploy=full_deploy`` deployer runs before ``CMakeDeps``,
+the files generated by ``CMakeDeps`` will point to the local copy in the user folder done by the ``full_deploy`` deployer,
```

Comment:
why do they need to affect on generators behavior? that's super confusing part.
as a consumer, I'd expect deployers just to deploy (copy) things, and otherwise, leave other parts untouched.

Reply:
That is one of most requested behaviors for deployers and generators.

Reply:
that's kinda surprising behavior that deployers may modify files generated by generators. can it be made more explicit? e.g. `--deploy=full_deploy --deploy-modify-generator=CMakeDeps`?